<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Editor::MainWindow</class>
 <widget class="QMainWindow" name="Editor::MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>664</width>
    <height>617</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowNestedDocks|QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="instrumentsSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="innerWidget">
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="Editor::Sequencer" name="sequencer" native="true"/>
          <widget class="Editor::PianoRoll" name="pianoRoll" native="true"/>
          <widget class="QSplitter" name="editorSplitter">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <widget class="Editor::MmlCommandPalette" name="mmlCommandPalette" native="true"/>
           <widget class="Editor::MmlCommandEditor" name="mmlCommandEditor" native="true"/>
           <widget class="Editor::MmlEditor" name="mmlEditor" native="true"/>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="Editor::SampleListWidget" name="instruments" native="true">
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>0</height>
        </size>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBarFile">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::TopToolBarArea</set>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionLoad"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QToolBar" name="toolBarPlayback">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::TopToolBarArea</set>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPlayPause"/>
   <addaction name="actionRewind"/>
   <addaction name="actionLoop"/>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>664</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionLoad"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuPlay">
    <property name="title">
     <string>&amp;Play</string>
    </property>
    <addaction name="actionPlayPause"/>
    <addaction name="actionRewind"/>
    <addaction name="separator"/>
    <addaction name="actionLoop"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>&amp;Window</string>
    </property>
    <addaction name="actionShowSequencer"/>
    <addaction name="actionShowCommandPalette"/>
    <addaction name="actionShowCommandEditor"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuPlay"/>
   <addaction name="menuWindow"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="../editor.qrc">
     <normaloff>:/images/24x24/song-new.png</normaloff>:/images/24x24/song-new.png</iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="toolTip">
    <string>Create a new song</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="icon">
    <iconset resource="../editor.qrc">
     <normaloff>:/images/24x24/song-load.png</normaloff>:/images/24x24/song-load.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Load</string>
   </property>
   <property name="toolTip">
    <string>Load a song</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../editor.qrc">
     <normaloff>:/images/24x24/song-save.png</normaloff>:/images/24x24/song-save.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="toolTip">
    <string>Save the song</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="icon">
    <iconset resource="../editor.qrc">
     <normaloff>:/images/24x24/song-export.png</normaloff>:/images/24x24/song-export.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Export</string>
   </property>
   <property name="toolTip">
    <string>Export the song</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionPlayPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../editor.qrc">
     <normaloff>:/images/24x24/song-play.png</normaloff>:/images/24x24/song-play.png</iconset>
   </property>
   <property name="text">
    <string>Play</string>
   </property>
   <property name="toolTip">
    <string>Play or Pause the song</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionRewind">
   <property name="icon">
    <iconset resource="../editor.qrc">
     <normaloff>:/images/24x24/song-rewind.png</normaloff>:/images/24x24/song-rewind.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Rewind</string>
   </property>
   <property name="toolTip">
    <string>Rewind the song from the start</string>
   </property>
   <property name="shortcut">
    <string>F6</string>
   </property>
  </action>
  <action name="actionLoop">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../editor.qrc">
     <normaloff>:/images/24x24/song-loop.png</normaloff>:/images/24x24/song-loop.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Loop</string>
   </property>
   <property name="toolTip">
    <string>Loops the song</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset resource="../editor.qrc">
     <normaloff>:/images/24x24/song-save.png</normaloff>:/images/24x24/song-save.png</iconset>
   </property>
   <property name="text">
    <string>Save &amp;as...</string>
   </property>
  </action>
  <action name="actionShowSequencer">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show &amp;Sequencer</string>
   </property>
  </action>
  <action name="actionShowCommandPalette">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Command &amp;Palette</string>
   </property>
  </action>
  <action name="actionShowCommandEditor">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Command &amp;Editor</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Editor::SampleListWidget</class>
   <extends>QWidget</extends>
   <header location="global">editor/sample/SampleListWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Editor::Sequencer</class>
   <extends>QWidget</extends>
   <header location="global">editor/sequencer/Sequencer.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Editor::PianoRoll</class>
   <extends>QWidget</extends>
   <header location="global">editor/piano/PianoRoll.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Editor::MmlEditor</class>
   <extends>QWidget</extends>
   <header location="global">editor/mml/MmlEditor.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Editor::MmlCommandPalette</class>
   <extends>QWidget</extends>
   <header location="global">editor/mml/MmlCommandPalette.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Editor::MmlCommandEditor</class>
   <extends>QWidget</extends>
   <header location="global">editor/mml/MmlCommandEditor.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../editor.qrc"/>
 </resources>
 <connections/>
</ui>
